trulens_eval/examples/vector-dbs/pinecone/constructing_optimal_pinecone.ipynb

```diff
@@ -0,0 +1,1516 @@
+     "data": {
+      "text/plain": [
+       "{'query': 'How many total major trophies has manchester united won?',\n",
+       " 'result': 'Manchester United has won a total of 66 major trophies.'}"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "✅ record record_hash_1e3663ba5074876b91386082ce0a9dc1 from Chain2_WikipediaQA -> default.sqlite\n",
+      "✅ feedback feedback_result_hash_3035174c65658e70bb444482ce7b7553 on record_hash_3961d1db52a3db2d762b8ee1cdb4131c -> default.sqlite\n",
+      "✅ feedback feedback_result_hash_6f19ade4a510f1eb47c1602202ff7b6e on record_hash_3961d1db52a3db2d762b8ee1cdb4131c -> default.sqlite\n",
+      "✅ feedback feedback_result_hash_75c9f85ab2e694000596bf10d6df4899 on record_hash_1e3663ba5074876b91386082ce0a9dc1 -> default.sqlite\n",
+      "✅ feedback feedback_result_hash_58252ee6bb8b914a6fff8ca37e9f4af8 on record_hash_1e3663ba5074876b91386082ce0a9dc1 -> default.sqlite\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "openai request failed <class 'openai.error.ServiceUnavailableError'>=The server is overloaded or not ready yet.. Retries=3.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "✅ feedback feedback_result_hash_333828424de0c468095e5c2eea55be55 on record_hash_d6ed647224bf101c6f4e265266c55a9c -> default.sqlite\n",
+      "✅ feedback feedback_result_hash_6f1a58d86712dce5a755a5c86be4235e on record_hash_d6ed647224bf101c6f4e265266c55a9c -> default.sqlite\n"
+     ]
+    }
+   ],
+   "source": [
+    "truchain(\"Name some famous dental floss brands?\")\n",
+    "truchain(\"Which year did Cincinatti become the Capital of Ohio?\")\n",
+    "truchain(\"Which year was Hawaii's state song written?\")\n",
+    "truchain(\"How many countries are there in the world?\")\n",
+    "truchain(\"How many total major trophies has manchester united won?\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pinecone.delete_index(index_name_v2)\n",
+    "time.sleep(30) # sleep for 30 seconds after deleting the index before creating a new one"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "index_name_v3 = 'langchain-rag-dot'\n",
+    "pinecone.create_index(\n",
+    "        name=index_name_v3,\n",
+    "        metric='dotproduct',\n",
+    "        dimension=1536,  # 1536 dim of text-embedding-ada-002\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'dimension': 1536,\n",
+       " 'index_fullness': 0.0,\n",
+       " 'namespaces': {},\n",
+       " 'total_vector_count': 0}"
```

Comment:
This is probably the only cell output that is useful to the tutorial

total_vector_count:0
and then after it's upserted

total_vector_count:30000

Let's just add a comment to look for that value and then clear the notebook cell outputs